starknet-foundry/crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -173,19 +169,16 @@ fn execute_syscall(
     blockifier_state: &mut CachedState<DictStateReader>,
 ) -> Result<(), HintError> {
     let (cell, offset) = extract_buffer(system);
-    let mut system_ptr = get_ptr(vm, cell, &offset)?;
+    let system_ptr = get_ptr(vm, cell, &offset)?;
 
-    let selector = felt_from_pointer(vm, &mut system_ptr)
-        .unwrap()
-        .to_bytes_be();
+    let mut buffer = MemBuffer::new(vm, system_ptr);
 
-    let gas_counter = usize_from_pointer(vm, &mut system_ptr).unwrap();
-    let contract_address = felt_from_pointer(vm, &mut system_ptr).unwrap();
-    let entry_point_selector = felt_from_pointer(vm, &mut system_ptr).unwrap();
+    let selector = buffer.next_felt252()?.to_bytes_be();
+    let gas_counter = buffer.next_usize()?;
+    let contract_address = buffer.next_felt252()?.into_owned();
+    let entry_point_selector = buffer.next_felt252()?.into_owned();
```

Comment:
Is propagating the error here ok, or should I always unwrap to not propagate memory errors?

Reply:
I think we decided that if the errors are never supposed to happen (unless the code is wrong) they should panic
cc @Arcticae @cptartur @MaksymilianDemitraszek

Reply:
if it's on our side of code, we should definetely panic since it's developer (our) error